### PR TITLE
New pages: HTMLInputElement step, name, and defaultValue properties

### PR DESCRIPTION
**Pull request by estelle**
part of https://github.com/mdn/content/issues/35544

### CODE REVIEW

File: files/en-us/web/api/htmlinputelement/defaultvalue/index.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+title: "HTMLInputElement: defaultValue property"
+short-title: defaultValue
+slug: Web/API/HTMLInputElement/defaultValue
+page-type: web-api-instance-property
+browser-compat: api.HTMLInputElement.defaultValue
+---
+
+{{ApiRef("HTML DOM")}}
+
+The **`defaultValue`** property of the {{domxref("HTMLInputElement")}} interface indicates the original (or default) value of the {{HTMLElement("input")}} element's [`value`](/en-US/docs/Web/HTML/Element/input#value) attribute as originally specified in the HTML that created the control.
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/defaultvalue/index.md:11]: Our standard lingo is "it reflects the `<input>` element's `value` attribute". No need to say "as originally specified...".

```suggestion
The **`defaultValue`** property of the {{domxref("HTMLInputElement")}} interface indicates the original (or default) value of the {{HTMLElement("input")}} element. It reflects the element's [`value`](/en-US/docs/Web/HTML/Element/input#value) attribute.
```

File: files/en-us/web/api/htmlinputelement/defaultvalue/index.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+## Value
+
+A string representing the default, or original, value of the `<input>` element.
+
+## Example
+
+This example demonstrates retrieving the original and current values of an input element.
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/defaultvalue/index.md:19]: TBH, I don't think this example is interesting enough to warrant a live sample, and I'd very much prefer it to be two lines like
```js
const inputElement = document.getElementById("planet");
console.log(inputElement.defaultValue); // "Azarath"
```
Just my preference though.

File: files/en-us/web/api/htmlinputelement/name/index.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+title: "HTMLInputElement: name property"
+short-title: name
+slug: Web/API/HTMLInputElement/name
+page-type: web-api-instance-property
+browser-compat: api.HTMLInputElement.name
+---
+
+{{ApiRef("HTML DOM")}}
+
+The **`name`** property of the {{domxref("HTMLInputElement")}} interface indicates the name of the {{HTMLElement("input")}} element, generally defined set by the element's [`name`](/en-US/docs/Web/HTML/Element/input#name) attribute. This attribute can also be provide an element with a different name.
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/name/index.md:11]:
```suggestion
The **`name`** property of the {{domxref("HTMLInputElement")}} interface indicates the name of the {{HTMLElement("input")}} element. It reflects the element's [`name`](/en-US/docs/Web/HTML/Element/input#name) attribute.
```

File: files/en-us/web/api/htmlinputelement/step/index.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+title: "HTMLInputElement: step property"
+short-title: step
+slug: Web/API/HTMLInputElement/step
+page-type: web-api-instance-property
+browser-compat: api.HTMLInputElement.step
+---
+
+{{ApiRef("HTML DOM")}}
+
+The **`step`** property of the {{domxref("HTMLInputElement")}} interface indicates the step of the {{HTMLElement("input")}} element, generally defined set by the numeric or date-time input element's [`step`](/en-US/docs/Web/HTML/Element/input#step) attribute. The `step` property can also be used to provide an element with a `step` value. Valid values include the string `any`, a positive integer, or a positive floating point number
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/step/index.md:11]:
```suggestion
The **`step`** property of the {{domxref("HTMLInputElement")}} interface indicates the step of the {{HTMLElement("input")}} element, generally defined set by the numeric or date-time input element's [`step`](/en-US/docs/Web/HTML/Element/input#step) attribute. It reflects the element's [`step`](/en-US/docs/Web/HTML/Element/input#step) attribute. Valid values include the string `"any"` or a string containing a positive floating point number.
```

File: files/en-us/web/api/htmlinputelement/step/index.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+The string value represents the element's `step` attribute, which works with `min` and `max` attributes, minumum and maximum values, to limit the increments at which a range, numeric or date-time value can be set. If not set to `any`, the control accepts only values at multiples of the step value greater than the minimum. If not set, the `step` property is an empty string.
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/step/index.md:13]: This should be documented in the HTML element page. There's nothing interesting to be noted about the IDL property other than it reflects the HTML attribute.

**Review by Josh-Cena — CHANGES_REQUESTED** [review]

**Comment by Josh-Cena** [comment]
In general, IDL properties are actually quite easy to document because all you have to say is "it reflects the attribute `blah`. For all valid values and what they do, read the HTML documentation instead", and then give a three-line example demonstrating reading, and occasionally writing, the property.

**Comment by estelle** [comment]
> give a three-line example demonstrating reading, and occasionally writing, the property.

Thanks @Josh-Cena. That will make this project much easier ;)

File: files/en-us/web/api/htmlinputelement/step/index.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+title: "HTMLInputElement: step property"
+short-title: step
+slug: Web/API/HTMLInputElement/step
+page-type: web-api-instance-property
+browser-compat: api.HTMLInputElement.step
+---
+
+{{ApiRef("HTML DOM")}}
+
+The **`step`** property of the {{domxref("HTMLInputElement")}} interface indicates the step of the {{HTMLElement("input")}} element, generally defined set by the numeric or date-time input element's [`step`](/en-US/docs/Web/HTML/Element/input#step) attribute. It reflects the element's [`step`](/en-US/docs/Web/HTML/Element/input#step) attribute. Valid values include the string `"any"` or a string containing a positive floating point number. If the attribute is not explicitly set, the `step` property is an empty string.
```
Comment (Josh-Cena) [inline files/en-us/web/api/htmlinputelement/step/index.md:11]:
```suggestion
The **`step`** property of the {{domxref("HTMLInputElement")}} interface indicates the step by which numeric or date-time {{HTMLElement("input")}} elements can change. It reflects the element's [`step`](/en-US/docs/Web/HTML/Element/input#step) attribute. Valid values include the string `"any"` or a string containing a positive floating point number. If the attribute is not explicitly set, the `step` property is an empty string.
```

**Review by Josh-Cena — APPROVED** [review]
Just one nit, but I think this is a pattern generalizable to most reflection properties.

**Pull request merged**